sstables/sstables.cc

```diff
@@ -3320,8 +3320,63 @@ gc_clock::time_point sstable::get_gc_before_for_fully_expire(const gc_clock::tim
     return res.knows_entire_range ? res.min_gc_before : gc_clock::time_point::min();
 }
 
+future<> remove_table_directory_if_has_no_snapshots(fs::path table_dir) {
+    // Be paranoid about risky paths
+    if (table_dir == "" || table_dir == "/") {
+        on_internal_error_noexcept(sstlog, format("Invalid table directory for removal: {}", table_dir));
+        abort();
+    }
+
+    // Return error code, 0 is success
+    auto remove_dir = [] (fs::path dir, bool recursive) -> future<int> {
```

Comment:
Nitpick: why make this a lambda instead of a normal static function?

Reply:
I'm reluctant to make it available to other users since it's quite specific to the calling function,
but we can make it a private static function at this point until we find other users for it.

Reply:
Unless I'm missing something, this simple function doesn't use anything from the class or anything. It can be an old-school "static" function (a file-local function). It doesn't need to be a "private static" function in the C++ classes sense.
But I don't insist. If you like this lambda thing, feel free to keep it.

Reply:
true, I'll make it a static free function then.
it may help readability of the code.